File: app-v3/src/main/java/com/khmelenko/lab/varis/repositories/search/SearchResultsActivityModule.java
package com.khmelenko.lab.varis.repositories.search;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

/**
 * @author Dmytro Khmelenko (d.khmelenko@gmail.com)
 */
@Module(subcomponents = SearchResultsActivitySubcomponent.class)
public abstract class SearchResultsActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(SearchResultsActivity.class)
    abstract AndroidInjector.Factory<? extends Activity>
    bindSearchResultsActivityInjectorFactory(SearchResultsActivitySubcomponent.Builder builder);
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/repositories/search/SearchResultsActivityModule.kt
package com.khmelenko.lab.varis.repositories.search

import android.app.Activity

import dagger.Binds
import dagger.Module
import dagger.Subcomponent
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap

@Module(subcomponents = arrayOf(SearchResultsActivitySubcomponent::class))
abstract class SearchResultsActivityModule {
    @Binds
    @IntoMap
    @ActivityKey(SearchResultsActivity::class)
    internal abstract fun bindSearchResultsActivityInjectorFactory(
            builder: SearchResultsActivitySubcomponent.Builder): AndroidInjector.Factory<out Activity>
}

@Subcomponent
interface SearchResultsActivitySubcomponent : AndroidInjector<SearchResultsActivity> {
    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<SearchResultsActivity>()
}
